src/features/channel-state.js

```diff
@@ -31,20 +31,22 @@ var resetCountDown = function() {
 var initiateCountDown = function(length) {
     if (bttv.chat.store.chatCountDown) clearInterval(bttv.chat.store.chatCountDown);
 
-    var timer = length;
+    var startTimestamp = Date.now();
+    var endTimestamp = startTimestamp + (length * 1000);
 
     bttv.chat.store.chatCountDown = setInterval(function() {
+        var currentTimestamp = Date.now();
+        var remainingTime = endTimestamp - currentTimestamp;
         var $chatButton = $(chatButton);
 
-        if (timer === 0) {
+        if (remainingTime <= 0) {
             resetCountDown();
             return;
+        } else {
```

Comment:
If you're returning in the above `if`, then the `else` block is superfluous. That's why it doesn't exist in the code you're editing.